cmd/async/async.go

```diff
@@ -0,0 +1,148 @@
+package main
+
+import (
+	"context"
+	"database/sql"
+	"database/sql/driver"
+	"flag"
+	"fmt"
+	sf "github.com/snowflakedb/gosnowflake"
+	"io"
+	"log"
+	"os"
+	"strconv"
+	"strings"
+)
+
+func getDSN() (string, *sf.Config, error) {
```

Comment:
I have seen similar code in other examples, is it possible to make a util func to do same/similar stuff?

Reply:
I believe that this should be in the scope of another issue as we have 11 duplicates of the function thus I have created a new issue for that: SNOW-832816.